keps/115-Subgroup-support/README.md

```diff
@@ -0,0 +1,232 @@
+
+
+This KEP is to have LeaderWorkerSet to support running both prefill/decode servers in one pod group (aka one replica) for LLM on multi-host
+
+
+## Proposal
+
+<!--
+This is where we get down to the specifics of what the proposal actually is.
+This should have enough detail that reviewers can understand exactly what
+you're proposing, but should not include things like API designs or
+implementation. What is the desired outcome and how do we measure success?.
+The "Design Details" section below is for the real
+nitty-gritty.
+-->
+
+### User Stories (Optional)
+
+<!--
+Detail the things that people will be able to do if this KEP is implemented.
+Include as much detail as possible so that people can understand the "how" of
+the system. The goal here is to make this feel real for users without getting
+bogged down.
+-->
+
+#### Story 1
+As a user, I should be able to split each LeaderWorkerSet replica into smaller subgroups, and be able to deploy each subgroup on a separate accelerator island. 
+
+### Risks and Mitigations
+
+<!--
+What are the risks of this proposal, and how do we mitigate? Think broadly.
+For example, consider both security and how this will impact the larger
+Kubernetes ecosystem.
+
+How will security be reviewed, and by whom?
+
+How will UX be reviewed, and by whom?
+
+Consider including folks who also work outside the SIG or subproject.
+-->
+To avoid the changes from affecting initial functionality, the default behavior 
+is to not set any labels, and for no subgroups to be created if `subGroupSize` is 
+not set.
+
+## Design Details
+
+The overall goal of subgroup scheduling
+
+<!--
+This section should contain enough information that the specifics of your
+change are understandable. This may include API specs (though not always
+required) or even code snippets. If there's any ambiguity about HOW your
+proposal will be implemented, this is the place to discuss them.
+-->
+### LeaderWorkerSet API
+We extend the LeaderWorkerSet API to introduce a new field: subGroupSize to opt in and set the number of pods that each subgroup will contain. Current behavior is kept if not set. 
+
+```
+type LeaderWorkerTemplate struct {
+	 // The number of pods per subgroup. This value is immutable,
+   // and must not be greater than LeaderWorkerSet.Spec.Size.
+   // Size must be divisible by subGroupSize in which case the 
+   // subgroups will be of equal size. Or size - 1 is divisible
+   // by subGroupSize, in which case the leader is considered as
+   // the extra pod, and will be part of the first subgroup.
+	SubGroupSize *int32 'json:"subGroupSize,omitempty'
```

Comment:
I think we should put this inside a `SubgroupingPolicy` struct, there is a good chance we will have more parameters related to that in the future (how to deal with the leader, subgroup exclusive policy)

Reply:
makes sense, added the struct